### ISSUE TITLE
Create cross-validation split on input dataset

### ISSUE THREAD

**Issue by pascalnotin** [OpenBioML/protein-lm-scaling#18]
1. Extract all Uniref30 cluster sequences from the input dataset and move them to test set.
2. Randomly split the remainder of the input dataset into train-val-test sets.
3. Combine the test sets from step 1 and 2 together.

**Comment by pascalnotin** [comment]
@jamaliki -- does that look good to you?

**Comment by jamaliki** [comment]
Yes this looks good to me @pascalnotin We can use colabfolddb so that if we want to use uniref100 later, we can do so easily.

**Comment by pascalnotin** [comment]
@jamaliki - I updated the description based on our latest discussion. Lmk your thoughts!

**Comment by jamaliki** [comment]
Makes sense @pascalnotin !

**Comment by csjackson0** [comment]
I can help on this if needed! @pascalnotin is there a Group1/Group2 split made up for ProteinGym?

**Comment by pascalnotin** [comment]
Thanks @csjackson0 ! I will create a split and share here over the weekend

**Comment by pascalnotin** [comment]
@csjackson0 -- apologies for the slight delay here. I think it will be best if the implementation of this issue takes as input two csv files: 'PG_test_sequences.csv' and 'PG_train_sequences.csv', where each file contains one reference sequence per line.

We will be releasing a much larger version of the ProteinGym benchmark in a few weeks, so this will help ensure we can re-run the corresponding script when this is out by just updating the two csv files as needed. 

For the time being, let's use the following:
- All sequences can be found in this mapping file: https://github.com/OATML-Markslab/ProteinGym/blob/main/ProteinGym_reference_file_substitutions.csv
- The 10 sequences in the 'PG_test_sequences.csv' file should be as follows:
A4D664_9INFA (Soh et al., 2019)
A4GRB6_PSEAI (Chen et al., 2020)
AMIE_PSEAE (Wrenbeck et al., 2017)
CALM1_HUMAN (Weile et al.,2017)
DLG4_RAT (McLaughlin Jr et al.,2012)
GAL4_YEAST (Kitzman et al.,2015)
PA_I34A1 (Wu et al.,2015)
Q2N0S59_HIV1 (Haddox et al.,2018)
SPG1_STRSG (Olson et al.,2014)
TPOR_HUMAN (Bridgford et al.,2020)
- All other sequences would go to  'PG_train_sequences.csv'

**Comment by csjackson0** [comment]
@pascalnotin Thanks for sending the sequences. Excited about the new ProteinGym benchmark! 

I was thinking of also adding the Uniref30 cluster file "uniref30_2302.tsv" to the input of the script.

**Comment by pascalnotin** [comment]
Hi @csjackson0 -- sounds good to use this file as an example to structure the code. All WT sequences from ProteinGym are in Uniref100, but not sure whether they will all be chosen as Uniref30 cluster representatives, so we may have to test for *approximate* match in case certain cluster representatives differ a bit from the WT seqs used in PG.
